sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilters.scala

```diff
@@ -48,6 +48,34 @@ private[sql] object ParquetFilters {
     override def inverseCanDrop(statistics: Statistics[T]): Boolean = false
   }
 
+  object StringFilter extends Enumeration {
+    type Mode = Value
+    val STARTS_WITH, ENDS_WITH, CONTAINS = Value
+  }
+
+  case class StringFilter(
+    v: java.lang.String,
```

Comment:
Scala `String` is just a type alias of `java.lang.String`, so we don't need to be explicit here.